test/functional/sqlsrv/sqlsrv_ae_datetimes_as_strings.phpt

```diff
@@ -0,0 +1,627 @@
+                }
+                break;
+            case 'datetime':
+                // Compare retrieved SQL datetime with expected date/time.
+                // SQL Server's datetime type is accurate to milliseconds and
+                // the expected time is accurate to microseconds, so append
+                // three zeroes to the retrieved time for comparison.
+                if ($retrievedDateTime."000" != $expectedDateTime[0] and
+                    $retrievedDateTime."000" != $expectedDateTime[1] and
+                    $retrievedDateTime."000" != $expectedDateTime[2] and
+                    $retrievedDateTime."000" != $expectedDateTime[3] ) {
+                        fatalError("Datetimes do not match!");
+                }
+                break;
+            case 'datetime2':
+                // Compare retrieved SQL datetime2 with expected date/time.
+                // SQL Server's datetime2 type is accurate to 100 ns and
+                // the expected time is accurate to microseconds, so append
+                // a zero to the expected time for comparison.
+                if ($retrievedDateTime != $expectedDateTime[0]."0" and
+                    $retrievedDateTime != $expectedDateTime[1]."0" and
+                    $retrievedDateTime != $expectedDateTime[2]."0" and
+                    $retrievedDateTime != $expectedDateTime[3]."0" ) {
+                        fatalError("Datetime2s do not match!");
+                }
+                break;
+            case 'datetimeoffset':
+                // Compare the SQL datetimeoffset retrieved with expected
+                // date/time. datetimeoffset is accurate to 100 ns, so the
+                // extra zero is removed in $dtoffset to create a format accurate 
+                // to microseconds for comparison with the expected date/time/timezone.
+                $dtoffset = $retrieved_date_time[0]." ".substr($retrieved_date_time[1], 0, -1)." ".$retrieved_date_time[2];
+                if ($dtoffset != $expectedDateTime[4] and
+                    $dtoffset != $expectedDateTime[5] and
+                    $dtoffset != $expectedDateTime[6] and
+                    $dtoffset != $expectedDateTime[7] ) {
+                        fatalError("Datetimeoffsets do not match!");
+                }
+                break;
+            case 'smalldatetime':
+                // Compare retrieved SQL smalldatetime with expected date/time.
+                // SQL Server's smalldatetime type is accurate to seconds and
+                // the expected time is accurate to microseconds, so append
+                // '.000000' to the expected time for comparison.
+                if ($retrievedDateTime.".000000" != $expectedDateTime[0] and
+                    $retrievedDateTime.".000000" != $expectedDateTime[1] and
+                    $retrievedDateTime.".000000" != $expectedDateTime[2] and
+                    $retrievedDateTime.".000000" != $expectedDateTime[3] ) {
```

Comment:
I thought smalldatetime is only accurate to minute (see line 326). Why is it being compared to the expectedDateTime with seconds precision (elements 0, 1 and 2)

Reply:
It is accurate to minute but may include the seconds (as ':00').

Reply:
then is should not be compared with $expectedDateTime[0], $expectedDateTime[1], and $expectedDateTime[2] right? Since they has seconds of 29.